REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/security/CustomPreAuthorize.java

```diff
@@ -0,0 +1,21 @@
+package com.epam.esm.security;
+
+import com.epam.esm.security.exception.SecurityAccessDeniedException;
+import com.epam.esm.user.entity.User;
+import com.epam.esm.user.entity.roles.Role;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service("securityService")
+@Slf4j
+public class CustomPreAuthorize {
+    public boolean customFilter(String id) {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        log.info("Requested id = {}, token id {}, token role {}", id, user.getId(), user.getRole());
+        if (!(user.getId().equals(Long.parseLong(id)) || user.getRole().equals(Role.ADMIN))) {
```

Comment:
I think you can add additionam method to User class to check role.
Instead of `user.getRole().equals(Role.ADMIN)` make it `user.is(Role.ADMIN)`

Reply:
Added isAdmin() method.
Why need to pass the role to method argument ?

Reply:
In case if you have more roles and more various checks.